### DISCUSSION TITLE
Solved: A problem with newlines in variable values on macOS

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#250] [Debugging and testing]
@gapmiss reported the following bug here: https://github.com/Taitava/obsidian-shellcommands/issues/1#issuecomment-1197307219
> ```shell
> # New Note from Clipboard
> # prompt modal for note title
> echo {{clipboard}} >> {{!_noteTitle}}.md
> open "obsidian://advanced-uri?vault=CiO&filepath={{!_noteTitle}}"
> ```
> Only issue found:
> 
> line-breaks(\n) do NOT get converted to actual line-breaks in the markdown
> 
> In the "New Note from Clipboard" command, with the Clipboard:
> 
> ```
> Shell Commands v0.14.0
> Obsidian v0.15.9 (installer version: v0.15.9)
> macOS Monterey 12.4 (21F79)
> ```
> 
> resulting markdown:
> 
> ```
> Shell Commands v0.14.0\nObsidian v0.15.9 (installer version: v0.15.9)\nmacOS Monterey 12.4 (21F79)\n
> ```

**Comment by Taitava** [comment]
@gapmiss May I ask what shell are you using: `Bash`, `Dash` or `Zsh`? [Here's how to find it out if you are not sure](https://publish.obsidian.md/shellcommands/Environments/Shells#How+to+know+which+shell+is+used). Thanks! 🙂

---

And just out of curiosity, in `>> {{!_noteTitle}}.md` you are using the exclamation mark `!` to prevent escaping the variable value, may I ask why? 🙂 I'm interested to learn about situations where people end up using the exclamation mark so that I can better understand what kind of problems people may face with the escaping. I know this `{{!_noteTitle}}` part does not relate to the actual newline problem, so my question is a bit offtopic. 🙂

**Reply by gapmiss** [reply]
I am using `bash`.

The `_noteTitle` is the results of a "Preaction" prompt. So, being confident that there is not line breaks or dangerous characters in the note title I assign, I skip the escaping. I hope that makes sense.

**Comment by Taitava** [comment]
# I ran into this too, on Linux

Sorry @gapmiss , I forgot this bug completely until encountering it myself.

Just some quick thoughts: At this point I think it might be related to `\n` being escaped as `\\n`. But I'm not certain if that's the cause of this bug, and don't have time to check it right now.

Some images just for myself when I start digging into this:
- [image1](https://user-images.githubusercontent.com/13002142/204101367-2a481d96-58e2-49df-98c8-68f73f683352.png)
- [image2](https://user-images.githubusercontent.com/13002142/204101398-6b61a2ce-11ac-41d4-8a99-6d51319531cc.png)

**Comment by Taitava** [comment]
# Possible solution?

@gapmiss can you plese try this:

Add argument `-e` to the `echo` command:
```
# New Note from Clipboard
# prompt modal for note title
echo -e {{clipboard}} >> {{!_noteTitle}}.md
open "obsidian://advanced-uri?vault=CiO&filepath={{!_noteTitle}}"
```

`-e` should make `echo` interpret `\n` as a real newline character. I had completely forgotten about this behavior of `echo`. I found an old issue where I had noticed that `-e` is needed: #109. 

If this works for you, I guess there's nothing to "fix" in SC. :+1:

**Reply by gapmiss** [reply]
Yes! The `-e` flag solves the issue. Appreciate your due diligence.

**Reply by Taitava** [reply]
Thank you! 🙂 Just too bad it took me a few months to solve as I didn't remember the `-e` option on July. 😓
